### PR TITLE
[csharp-netcore] [HttpClient] Refactored Models

### PR DESCRIPTION
**Pull request by devhl-labs**
One typo fixed in RestSharp library.

Refactored model constructor to use null reference types. A model's ToJson method now takes an optional JsonSerializerSettings object which defaults to the static object in ClientUtils when not set. Removed a model's GetHashCode override as this should probably be specifically set by end users.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:

  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh 
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/csharp-netcore/libraries/httpclient/README.mustache [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+{{#netStandard}}
+- .NET Core >=1.0
+- .NET Framework >=4.6
+- Mono/Xamarin >=vNext
+{{/netStandard}}
+
+<a name="dependencies"></a>
+## Dependencies
+
+- [RestSharp](https://www.nuget.org/packages/RestSharp) - 106.11.4 or later
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/csharp-netcore/libraries/httpclient/README.mustache:62]: @devhl-labs Don't think we need RestSharp as the dependency

**Comment by devhl-labs** [comment]
Removed that RestSharp reference and addressed the issues from @Blackclaws.

**Comment by wing328** [comment]
@devhl-labs can you please try the latest master? We've added some tests, fixed a few minor issues and looks like the `httpclient` option is ready for some more tests.

Can you please give it a try to see if it meets your requirement?

**Comment by devhl-labs** [comment]
I'm confused, I only see ApiClient.mustache in the HttpClient folder. Did you intend for me to add my templates to master and do a new pr?

**Comment by devhl-labs** [comment]
I did a new pr to master. 

https://github.com/OpenAPITools/openapi-generator/pull/9073

**Comment by devhl-labs** [comment]
I will revisit this branch once #9235 is merged.

**Pull request closed without merging**